mmdeploy/codebase/mmdet3d/deploy/voxel_detection.py

```diff
@@ -0,0 +1,263 @@
+            ann_info=dict(axis_align_matrix=np.eye(4)),
+            sweeps=[],
+            # set timestamp = 0
+            timestamp=[0],
+            img_fields=[],
+            bbox3d_fields=[],
+            pts_mask_fields=[],
+            pts_seg_fields=[],
+            bbox_fields=[],
+            mask_fields=[],
+            seg_fields=[])
+        data = test_pipeline(data)
+        data = collate([data], samples_per_gpu=1)
+        data['img_metas'] = [
+            img_metas.data[0] for img_metas in data['img_metas']
+        ]
+        data['points'] = [point.data[0] for point in data['points']]
+        if device != 'cpu':
+            data = scatter(data, [device])[0]
+        else:
+            data['img_metas'] = data['img_metas'][0]
+            data['points'] = data['points'][0]
+        return data
+
+    @staticmethod
+    def run_inference(model, model_inputs: Dict[str, torch.Tensor]):
+        """Run inference once for a object detection model of mmdet3d.
+
+        Args:
+            model (nn.Module): Input model.
+            model_inputs (dict): A dict containing model inputs tensor and
+                meta info.
+
+        Returns:
+            list: The predictions of model inference.
+        """
+        result = model(
+            return_loss=False,
+            points=model_inputs['points'],
+            img_metas=model_inputs['img_metas'])
+        return [result]
+
+    @staticmethod
+    def evaluate_outputs(model_cfg,
+                         outputs: Sequence,
+                         dataset: Dataset,
+                         metrics: Optional[str] = None,
+                         out: Optional[str] = None,
+                         metric_options: Optional[dict] = None,
+                         format_only: bool = False,
+                         log_file: Optional[str] = None):
+        if out:
+            logger = get_root_logger()
+            logger.info(f'\nwriting results to {out}')
+            mmcv.dump(outputs, out)
+        kwargs = {} if metric_options is None else metric_options
+        if format_only:
+            dataset.format_results(outputs, **kwargs)
+        if metrics:
+            eval_kwargs = model_cfg.get('evaluation', {}).copy()
+            # hard-code way to remove EvalHook args
+            for key in [
+                    'interval', 'tmpdir', 'start', 'gpu_collect', 'save_best',
+                    'rule'
+            ]:
+                eval_kwargs.pop(key, None)
+                eval_kwargs.pop(key, None)
+            eval_kwargs.update(dict(metric=metrics, **kwargs))
+            print(dataset.evaluate(outputs, **eval_kwargs))
+
+    def get_model_name(self) -> str:
+        assert 'type' in self.model_cfg.model, 'model config contains no type'
+        name = self.model_cfg.model.type.lower()
+        return name
+
+    def get_tensor_from_input(self, input_data: Dict[str, Any],
+                              **kwargs) -> torch.Tensor:
+        pass
```

Comment:
for unimplemented method, could raise NotImplementedError if it is intended as so.